app/core/builder/blueprint.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+# For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+from typing import FrozenSet, Union, List, Iterable
+
+from astropy.time import TimeDelta
+import astropy.units as u
+
+from lucupy.minimodel.observation import ObservationClass
+from lucupy.minimodel.program import ProgramTypes
+from lucupy.minimodel.semester import Semester, SemesterHalf
+from lucupy.minimodel.site import Site, ALL_SITES
+
+from app.config import config, ConfigurationError
+from app.core.components.optimizer.dummy import DummyOptimizer
+
+class Blueprint:
+    """Base class for Blueprint
+    """
+    pass
+
+class CollectorBlueprint(Blueprint):
+    """Blueprint for the Collector.
+    This is based on the configuration in config.yml.
+    """
+    def __init__(self,
+                 semesters: str,
+                 obs_class: str,
+                 prg_type: str,
+                 sites: str,
+                 time_slot_length: float) -> None:
+        self.semesters: FrozenSet[Semester] =  frozenset(map(CollectorBlueprint._parse_semesters, semesters))
+        self.obs_classes: FrozenSet[ObservationClass] = frozenset(map(CollectorBlueprint._parse_obs_class, obs_class))
+        self.program_types: FrozenSet[ProgramTypes] = frozenset(map(CollectorBlueprint._parse_prg_types,prg_type))
+        self.sites: FrozenSet[Site] = CollectorBlueprint._parse_sites(sites)
+        self.time_slot_length: TimeDelta = TimeDelta(time_slot_length * u.min)
```

Comment:
It implicitly creates a constructor for the superclass... but since you're not implementing it, it doesn't really do anything such that we need to call it. I'm guilty of doing the same thing plenty of times.